    1  ls -al
    2  pwd 
    3  ls /mnt/
    4  cd /mnt/c
    5  clear
    6  ls -al
    7  ls
    8  lz
    9  cd /etc
   10  echo $SHELL
   11  which bash
   12  ls
   13  nano myscript.sh
   14  mkdir -p ~/miniconda3
   15  wget <https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh> -O ~/miniconda3/miniconda.sh
   16  bash ~/miniconda3/miniconda.sh -b -u -p ~/miniconda3
   17  conda create --name my_env
   18  conda --version
   19  nano myscript.sh
   20  sudo chmod +x myscript.sh
   21  ls -l
   22  ./myscript.sh
   23  cat myscript.sh
   24  nano myscript.sh
   25  pwd
   26  nano myscript.sh
   27  ./myscript.sh
   28  nano myscript.sh
   29  ./myscript.sh
   30  ls
   31  nano myscript.sh
   32  ./myscript.sh
   33  nano myscript.sh
   34  myname="Friday"
   35  echo "Hello world"
   36  echo $myname
   37  echo myname
   38  echo &newvariable
   39  echo $myname
   40  ls="hello again"
   41  ls
   42  echo $ls
   43  nano myscript.sh
   44  ./myscript.sh
   45  nano myscript.sh
   46  bash
   47  conda --version
   48  cd Downloads
   49  ls
   50  chmod +x Miniconda3-latest-Linux-x86_64.sh
   51  cd C:\Users\JHAIR\OneDrive - Universidad del Pacífico\Documentos\Descargas
   52  cd Descargas
   53  mv Downloads/Miniconda3-latest-Linux-x86_64.sh ~
   54  cd \Users\JHAIR\OneDrive - Universidad del Pacífico\Documentos\Descargas
   55  conda --version
   56  cd C:\Users\JHAIR\OneDrive - Universidad del Pacífico\Documentos\Descargas
   57  cd C:\\
   58  cd C:chmod +x Miniconda3-latest-Linux-x86_64.sh
   59  ./Miniconda3-latest-Linux-x86_64.sh
   60  cd C:\Users\JHAIR\miniconda3
   61  cdC:\Users\JHAIR\miniconda3
   62  cd C:\Users\JHAIR
   63  mv Downloads/Miniconda3-latest-Linux-x86_64.sh ~
   64  mv Descargas/Miniconda3-latest-Linux-x86_64.sh ~
   65  mv C:\Users\JHAIR\OneDrive - Universidad del Pacífico\Documentos\Descargas/Miniconda3-latest-Linux-x86_64.sh ~
   66  cd ~
   67  cd ~/projects
   68  cd ~/home
   69  cd ~/fridayv
   70  nano
   71  cd
   72  mkdir proyectos
   73  ls
   74  mv Descargas/Miniconda3-latest-Linux-x86_64.sh ~
   75  mv C:\Users\JHAIR\OneDrive - Universidad del Pacífico\Documentos\Descargas/Miniconda3-latest-Linux-x86_64.sh ~
   76  mv C:\Windows/Miniconda3-latest-Linux-x86_64.sh ~
   77  mv Windows/Miniconda3-latest-Linux-x86_64.sh ~
   78  ~
   79  cd
   80  cd ~
   81  bash ~/Descargas/Miniconda3-latest-Linux-x86_64.sh
   82  /mnt/c
   83  cd Users/JHAIR/OneDrive\ -\ Universidad\ del\ Pacífico/Documentos/Descargas
   84  cd C:\Users\JHAIR\OneDrive - Universidad del Pacífico\Documentos\Descargas
   85  ls
   86  /mnt/c
   87  cd Users\JHAIR\OneDrive - Universidad del Pacífico\Documentos\Descargas
   88  cp Miniconda3-latest-Linux-x86_64.sh .
   89  /mnt/c
   90  cd Descargas
   91  cd Users/JHAIR/Descargas
   92  cd cp Miniconda3-latest-Linux-x86_64.sh .
   93  cd "Users/JHAIR/OneDrive - Universidad del Pacífico/Documentos/Descargas"
   94  cd Users
   95  ls
   96  /mnt/c
   97  ls
   98  wget -O aconda.sh https://repo.anaconda.com/archive/Anaconda3-2024.02-1-Linux-x86_64.sh
   99  bash aconda.sh
  100  conda info
  101  conda --version
  102  conda
  103  anaconda --version
  104  Anaconda3 --version
  105  cd
  106  echo Hello
  107  vim textfile.txt
  108  mkdir charles darwin
  109  ~
  110  charles darwin
  111  cd charles darwin
  112  mkdir charles_darwin
  113  cd charles_darwin
  114  cd
  115  mkdir biocomputing && cd biocomputing
  116  wget https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.fna
  117  wget https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.gbk
  118  mv wildtype.fna Charles_Darwin/
  119  mv wildtype.fna charles_darwin/
  120  mv wildtype.fna charles_darwin
  121  wget https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.fna
  122  mv wildtype.fna /charles_darwin
  123  cd
  124  cd charles_darwin
  125  ls
  126  cd
  127  cd charles_darwin
  128  ~
  129  cd
  130  cd biocomputing
  131  mv wildtype.fna ~/charles_darwin
  132  rm wildtype.gbk
  133  mv wildtype.fna ~/biocomputing
  134  cd
  135  cd charles_darwin
  136  mv wildtype.fna ~/biocomputing
  137  cd
  138  cd computing
  139  cd biocomputing
  140  cat wildtype.fna
  141  grep "TATA" wildtype.fna
  142  grep "tata" wildtype.fna
  143  grep "tatata" wildtype.fna
  144  grep "tatatata" wildtype.fna > mutant_lines.txt
  145  cat mutant_lines.txt
  146  grep "tatata" mutant_lines.txt
  147  echo "BRCA1"
  148  curl -o BRCA1.fasta "https://www.ncbi.nlm.nih.gov/nuccore/NM_007294.4?report=fasta"
  149  wget https://www.ncbi.nlm.nih.gov/nuccore/NM_007294.4?report=fasta&log$=seqview&format=text
  150  wget -O BRCA1.fasta https://www.ncbi.nlm.nih.gov/nuccore/NM_007294.4?report=fasta&log$=seqview&format=text
  151  wget -0 BRCA1.fasta "https://www.ncbi.nlm.nih.gov/nuccore/NM_007294.4?report=fasta&log$=seqview&format=text"
  152  wget -O BRCA1.fasta "https://www.ncbi.nlm.nih.gov/nuccore/NM_007294.4?report=fasta&log$=seqview&format=text"
  153  wget -O BRCA1.fasta "https://www.ncbi.nlm.nih.gov/nuccore/NM_007294.4?rep
  154  ort=fasta&log$=seqview&format=text"
  155  wget -O BRCA12.fasta "https://www.ncbi.nlm.nih.gov/nuccore/NM_007294.4?report=fasta&log$=seqview&format=text"
  156  curl -o BRCA1.fasta "https://www.ncbi.nlm.nih.gov/nuccore/NM_007294.4?report=fasta"
  157  wget -O BRCA1.fasta "https://www.ncbi.nlm.nih.gov/nuccore/NM_007294.4?report=fasta&to=7088"
  158  curl -L "https://www.ncbi.nlm.nih.gov/nuccore/NM_007294.4?report=fasta" -o BRCA.fasta
  159  wget https://www.ncbi.nlm.nih.gov/nuccore/NM_007294.4?report=fasta&log$=seqview&format=text
  160  wget -O BRCA.txt "https://www.ncbi.nlm.nih.gov/nuccore/NM_007294.4?report=fasta&log$=seqview&format=text"
  161  cd
  162  conda install -c bioconda ncbi-genome-download
  163  esearch -db nucleotide -query "NC_030850.1" | efetch -format fasta > NC_030850.1.fasta
  164  source ./install-edirect.sh
  165  post -db nuccore -input <file_with_accs> -format acc   | efetch -format fasta > sequences.fasta
  166  sh -c "$(curl -fsSL https://ftp.ncbi.nlm.nih.gov/entrez/entrezdirect/install-edirect.sh)"
  167  export PATH=${HOME}/edirect:${PATH}
  168  epost -db nuccore -input NM_007294 -format acc | efetch -format fasta > BRCA1.fasta
  169  esearch -db protein -query "lycopene cyclase" |   efetch -format fasta
  170  cd biocomputing
  171  cd
  172  cd biocomputing
  173  epost -db nuccore -input <file_with_accs> -format acc | efetch -format fasta > sequences.fasta
  174  epost -db nuccore -input <file_with_accs> -format acc \| efetch -format fasta > sequences.fasta
  175  epost -db nuccore -input <NM_007294> -format acc \| efetch -format fasta > sequences.fasta
  176  esearch -db protein -query "lycopene cyclase" | }
  177  esearch -db gene -query "NM_007294" | > sequences.fasta
  178  -bash
  179  esearch -db gene -query "NM_007294" | sequences.fasta
  180  esearch -db nucleotide -query "NC_030850.1" | efetch -format fasta > NC_030850.1.fasta
  181  esearch -db nucleotide -query "NM_007294.4" | efetch -format fasta > BRCA.fasta
  182  esearch -db nucleotide -query "XM_063696027.1" | efetch -format fasta > p53.fasta
  183  grep -c -v "^>" p53.fasta
  184  tail -n +2 nombre_del_archivo.fasta | grep -o -i "A" | wc -l
  185  tail -n +2 p53.fasta | grep -o -i "A" | wc -l
  186  tail -n +2 p53.fasta | grep -o -i "G" | wc -l
  187  n_gc=$(grep -o -i "[GC]" nombre_del_archivo.fasta | wc -l)
  188  n_gc=$(grep -o -i "[GC]" p53.fasta | wc -l)
  189  total_length=$(grep -v "^>" nombre_del_archivo.fasta | tr -d "\n" | wc -c)
  190  gc_content=$(echo "scale=2; ($n_gc / $total_length) * 100" | bc)
  191  total_length=$(grep -v "^>" p53.fasta | tr -d "\n" | wc -c)
  192  gc_content=$(echo "scale=2; ($n_gc / $total_length) * 100" | bc)
  193  sudo apt update
  194  sudo apt install bc
  195  total_length=$(grep -v "^>" p53.fasta | tr -d "\n" | wc -c)
  196  gc_content=$(echo "scale=2; ($n_gc / $total_length) * 100" | bc)
  197  echo "El porcentaje de contenido de GC es: $gc_content%"
  198  touch YourName.nucleotide
  199  >>
  200  cd biocomputing
  201  n_gc=$(grep -o -i "[GC]" p53.fasta | wc -l)
  202  n_gc=$(grep -o "[GC]" p53.fasta | wc -l)
  203  total_length=$(grep -v "^>" p53 | tr -d "\n" | wc -c)
  204  total_length=$(grep -v "^>" p53.fasta | tr -d "\n" | wc -c)
  205  gc_content=$(echo "scale=2; ($n_gc / $total_length) * 100" | bc)
  206  echo "Percentage of GC is: $gc_content%"
  207  gc_content=$(awk '/^[^>]/ {total+=length($0); gc+=gsub(/[GCgc]/,"")} END {printf "%.2f%%\n", (gc/total)*100}' $p53.fasta)
  208  nano Abner_nucleotides.fasta
  209  cd biocomputing
  210  cd fridayv
  211  cd
  212  cd biocomputing
  213  nano Abner_nucleotides.fasta
  214  echo Abner_nucleotides.fasta
  215  nano Abner_nucleotides.fasta
  216  chmod +x Abner_nucleotides.sh
  217  ./Abner_nucleotides.sh
  218  ./Abner_nucleotides.fasta
  219  chmod +x Abner_nucleotides.fasta
  220  ./Abner_nucleotides.fasta
  221  nano Abner_nucleotides.fasta
  222  /Abner_nucleotides.fasta
  223  nano Abner_nucleotides.fasta
  224  ./Abner_nucleotides.fasta
  225  nano Abner_nucleotides.fasta
  226  cd
  227  cd biocomputing
  228  nano Abner.fasta
  229  chmod Abner.fasta
  230  chmod +x Abner.fasta
  231  ./Abner.fasta
  232  nano Abner.fasta
  233  git clone <https://github.com/Bukola19/Charles-Darwin.git>
  234  cd
  235  git init
  236  git status
  237  git add .
  238  cd biocomputing
  239  mkdir /output
  240  mkdir "/output"
  241  mkdir output
  242  mv Abner.fasta ~/output
  243  wget https://raw.githubusercontent.com/Bukola19/Charles-Darwin/main/Abner.fasta
  244  git init
  245  git status
  246  git add .
  247  git commit -m "First commit"
  248  gh repo clone Bukola19/Charles-Darwin
  249  git remote add origin https://github.com/Bukola19/Charles-Darwin.git
  250  git push -u origin master
  251  git push -u origin output
  252  git push -u origin master
  253  nano Abner.sh
  254  git checkout -b master
  255  git add .
  256  git commit -m "Tu mensaje de commit"
  257  git config --global user.email "abneer.vidaurre@upch.pe"
  258  git config --global user.name "FridayV"
  259  git push -u origin master
  260  git commit -m "Tu mensaje de commit"
  261  git remote add origin https://github.com/Bukola19/Charles-Darwin.git
  262  git push -u origin master
  263  clear
  264  nano Abner.sh
  265  chmod -x Abner.sh
  266  ./Abner.sh
  267  chmod +x Abner.sh
  268  ./Abner.sh
  269  nano Abner.sh
  270  cd
  271  mkdir prueba
  272  cd prueba
  273  clear
  274  mkdir abner
  275  mkdir biocomputing && cd biocomputing
  276  wget https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.fna https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.gbk
  277  https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.gbk
  278  mv wildtype.fna ~/abner
  279  rm *.gbk
  280  grep -o 'tatatata' wildtype.fna | wc -l
  281  cd
  282  cd prueba
  283  cd abner
  284  grep -o 'tatatata' wildtype.fna | wc -l
  285  cd biocomputing
  286  cd
  287  cd prueba && cd biocomputing
  288  wget https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.fna https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.gbk
  289  https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.gbk
  290  mv *.fna ~/abner
  291  mv *.fna ~/prueba/abner
  292  rm wildtype.gbk
  293  cd prueba
  294  cd
  295  cd prueba && cd abner
  296  grep -o 'tatatata' wildtype.fna | wc -l
  297  grep "tatatata" wildtype.fna > mutant_lines.txt
  298  echo “p53”
  299  esearch -db nucleotide -query "XM_063696027.1" | efetch -format fasta > p53.fasta
  300  grep -c -v "^>" p53.fasta
  301  tail -n +2 p53.fasta | grep -o -i "A" | wc -l
  302  tail -n +2 p53.fasta | grep -o -i "G" | wc -l
  303  tail -n +2 p53.fasta | grep -o -i "C" | wc -l
  304  tail -n +2 p53.fasta | grep -o -i "T" | wc -l
  305  n_gc=$(grep -o "[GC]" p53.fasta | wc -l)
  306  total_length=$(grep -v "^>" p53.fasta | tr -d "\n" | wc -c)
  307  gc_content=$(echo "scale=2; ($n_gc / $total_length) * 100" | bc)
  308  echo "Percentage of GC is: $gc_content%"
  309  nano Abner.fasta
  310  touch Abner.fasta
  311  nano Abner.fasta
  312  echo "Number of A: $(grep -o 'A' Abner.fasta | wc -l)" >> Abner.fasta
  313  echo "Number of A: $(grep -o 'A' Abner.fasta | wc -l)"
  314  echo "Number of G: $(grep -o 'G' Abner.fasta | wc -l)"
  315  echo "Number of T: $(grep -o 'T' Abner.fasta | wc -l)"
  316  echo "Number of C: $(grep -o 'C' Abner.fasta | wc -l)"
  317  nano Abner.fasta
  318  history > Abner.sh
