    1  df
    2  df -H
    3  mkdir Bukola
    4  mkdir "Bukola"
    5  mkdir Biocomputing
    6  cd Biocomputing
    7  wget https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.fna 
    8  wget https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.gbk
    9  wget https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.gbk 
   10  mv wildtype.fna "Bukola"/
   11  mv wildtype.fna $"Bukola"/
   12  clear
   13  ls
   14  mv wildtype.fna ../Bukola/
   15  ls
   16  rm wildtype.gbk
   17  ls
   18  VIM
   19  mkdir Bukola
   20  pwd
   21  ls
   22  wgethttps://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.fna 
   23  https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.gbk
   24  https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.gbk 
   25  wget https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.gbk
   26  https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.gbk 
   27  wget https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.gbk
   28  https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.gbk 
   29  wget https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.fna 
   30  wget https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.gbk
   31  https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.gbk 
   32  grep -i "tata" wildtype.fna
   33  grep -o "tata" wildtype.fna
   34  grep -o "tatatata" wildtype.fna
   35  grep "tatatata" wildtype.fna > mutant_lines.txt
   36  ls mutant_lines.txt
   37  cat mutant_lines.txt
   38  wget https://www.ncbi.nlm.nih.gov/nuccore/NC_000017.11?report=fasta&from=43044295&to=43170327&strand=true > BRCA1.fasta
   39  cat BRCA1.fasta
   40  wget "https://www.ncbi.nlm.nih.gov/nuccore/NC_000017.11?report=fasta&from=43044295&to=43170327&strand=true"
   41  cat NC_000017.11?report=fasta&from=43044295&
   42  wget -o "https://www.ncbi.nlm.nih.gov/nuccore/NC_000017.11?report=fasta&from=43044295&to=43170327&strand=true"|.BRCA1.fna
   43  wget -o "https://www.ncbi.nlm.nih.gov/nuccore/NC_000017.11?report=fasta&from=43044295&to=43170327&strand=true" | >BRCA1.fna
   44  wget https://www.ncbi.nlm.nih.gov/nuccore/NC_000017.11
   45  cat NC_000017.11
   46  wget -o https://www.ncbi.nlm.nih.gov/nuccore/NC_000017.11?report=fasta&from=43044295&to=43170327&strand=true 
   47  wget -o gene.fasta "https://www.ncbi.nlm.nih.gov/nuccore/NC_000017.11?report=fasta&from=43044295&to=43170327&strand=true"
   48  wget https://www.ncbi.nlm.nih.gov/nuccore/NC_000017.11?report=fasta&log$=seqview&format=text&from=43044295&to=43170327&strand=true
   49  wget "https://www.ncbi.nlm.nih.gov/nuccore/NC_000017.11?report=fasta&from=43044295&to=43170327&strand=true "
   50  ls
   51  head -n2 wget-log
   52  head -n5 gene.fasta
   53  head -n5 BRCA1.fasta
   54  cd biocomputing
   55  cd Biocomputing
   56  ls BRAC1.fasta
   57  ls BRCA1.fasta
   58  clear
   59  wget https://www.ncbi.nlm.nih.gov/nuccore/NC_000017.11?report=fasta&from=43044295&to=43170327&strand=true
   60  head eget-log
   61  head wget
   62  head wget-log
   63  mkdir Bukola
   64  wget -O breastcancer.gene "https://www.ncbi.nlm.nih.gov/sviewer/viewer.fcgi?db=nuccore&report=fasta&id=NC_000017.1"
   65  grep 2n-~
   66  grep -v ">" | wc -l
   67  grep -v ">" *.gene | head -n -1 | wc -l
   68  ls
   69  grep *.gene | head -n -1 | wc -l
   70  cat breastcancer.gene
   71  wc -l breastcancer.gene
   72  grep -v ">" breastcancer.gene | wc -l
   73  grep -v ">" breastcancer.gene | tr -d '\n' | grep -o 'A' | wc -l
   74  grep -v '>' breastcancer.gene | grep -o 'A' | wc -l
   75  grep -v '>' breastcancer.gene | grep -o 'G' | wc -l
   76  grep -v '>' breastcancer.gene | grep -o 'A' | wc -c
   77  grep -v '>' breastcancer.gene | grep -o 'G' | wc -c
   78  history
   79  wget -o GyrBgene.fasta "https://www.ncbi.nlm.nih.gov/sviewer/viewer.fcgi?db=nuccore&report=fasta&id=PP072261.1
   80  "
   81  ls
   82  cat GyrBgene.fasta
   83  ls
   84  wget -0 GyrBgene2.fasta "https://www.ncbi.nlm.nih.gov/sviewer/viewer.fcgi?db=nuccore&report=fasta&id=PP072261.1"
   85  wget -o GyrBgene2.fasta "https://www.ncbi.nlm.nih.gov/sviewer/viewer.fcgi?db=nuccore&report=fasta&id=PP072261.1"
   86  ls
   87  cat GyrBgene2.fasta 
   88  wget -O GyrBgene2.fasta "https://www.ncbi.nlm.nih.gov/sviewer/viewer.fcgi?db=nuccore&report=fasta&id=PP072261.1"
   89  ls
   90  cat GyrBgene2.fasta 
   91  wc -l GyrBgene2.fasta 
   92  grep -v ">" GyrBgene2.fasta | wc -l
   93  $ grep -v ">" GyrBgene2.fasta
   94  grep -v ">" GyrBgene2.fasta
   95  $ grep -v ">" GyrBgene2.fasta | grep -o "A"
   96  grep -v ">" GyrBgene2.fasta | grep -o "A"
   97  grep -v ">" GyrBgene2.fasta | grep -o "A" | wc
   98  grep -v ">" GyrBgene2.fasta | grep -o "A" | wc -l
   99  grep -v '>' GyrBgene2.fasta | grep -o 'G' | wc -l
  100  grep -v '>' GyrBgene2.fasta | grep -o 'C' | wc -l
  101  grep -v '>' GyrBgene2.fasta | grep -o 'T' | wc -l
  102  GC_count=$(grep -o -i '[GC]' GyrBgene2.fasta | wc -l)
  103  grep -o -i '[GC]' hemoglobin_beta.fasta | wc -l | > GCcount)
  104  grep -o -i '[GC]' GyrBgene.fasta | wc -l | > GCcount
  105  cat GCcount 
  106  cat GCcount
  107  GC_count=$(grep -o -i '[GC]' GyrBgene2.fasta | wc -l)
  108  ls
  109  cat GCcount 
  110  grep -o '[GC]' GyrBgene.fasta | wc -l | > GCcount
  111  cat GCcount 
  112  touch Bukola.fasta
  113  ls
  114  rm BRCA1.fasta
  115  ls
  116  rm BRCA1.fna | rm breastcancer.gene
  117  ls
  118  rm 'NC_000017.11?report=fasta'
  119  echo $GCcount
  120  rm 'NC_000017.11?report=fasta'
  121  help
  122  ls
  123  GC-Count=$(grep -v '>' GyrBgene2.fasta |grep -o [GC] |wc -l
  124  )
  125  echo GC-Count
  126  echo $GC-Count
  127  GC_Count=$(grep -v '>' GyrBgene2.fasta |grep -o [GC] |wc -l
  128  )
  129  echo $GC_Count
  130  Total_Count=$(grep -v '>' GyrBgene2.fasta |grep -o [AGTC] |wc -l)
  131  echo $Total_Count
  132  cat GC_Count
  133  cat $GC_Count
  134  history
  135  ls
  136  grep -v '>'GyrBgene2.fasta | wc -l
  137  grep -v '>' GyrBgene2.fasta | wc -l
  138  grep -v '>' GyrBgene2.fasta | tail -n | wc -l
  139  grep -v '>' GyrBgene2.fasta | tail -n1 | wc -l
  140  grep -v '>' GyrBgene2.fasta | tail n -1 | wc -l
  141  grep -v '>' GyrBgene2.fasta | tail --n1 | wc -l
  142  Total_Count=$(grep -v '>' GyrBgene2.fasta |wc -c)
  143  echo $Total_Count
  144  Total_Count=$(grep -v '>' GyrBgene2.fasta |grep -o [AGTC] |wc -l)
  145  echo $Total_Count
  146  cat GyrBgene2.gene
  147  cat GyrBgene.fasta 
  148  (grep -v '^>' GyrBgene2.fasta |wc -c)
  149  Total_Count=$(grep -v '>' GyrBgene2.fasta |grep -o [AGTC] |wc -c)
  150  echo $Total_Count
  151  Total_Count=$(grep -v '>' GyrBgene2.fasta |grep -o [AGTC] |wc -l)
  152  echo $Total_Count
  153  history
  154  GC_Content= ($GC_Count/$Total_Count * 100 | bc)
  155  GC_Content= $ ($GC_Count/$Total_Count * 100 | bc)
  156  GC_Content= $ (($GC_Count/$Total_Coun)t * 100 | bc)
  157  GC_Content= $ (($GC_Count/$Total_Count) * 100 | bc)
  158  GC_Content=$(($GC_Count/$Total_Count) * 100 | bc)
  159  GC_Content=$("$GC_Count/$Total_Count) * 100" | bc)
  160  GC_Content=$"($GC_Count/$Total_Count) * 100" | bc)
  161  GC_Content=$("Scale=2; ($GC_Count/$Total_Count) * 100" | bc)
  162  GC_Content=$("Scale=2;$GC_Count / $Total_Count) * 100" | bc)
  163  GC_Content=$("Scale=2 ;$GC_Count / $Total_Count) * 100" | bc)
  164  (echo "scale=2; ($GC_Count / $Total_Count) * 100" | bc)
  165  (echo "scale=2; ($GC_Count / $Total_Count) * 100")
  166  -bash: bc: command not found
  167  bukkyomotore@cloudshell:~$
  168  echo $GC_Content
  169  ls
  170  touch Bukola.fasta
  171  echo "Number of A: $(grep -v '>' GyrBgene2.fasta | grep -o 'A' | wc -l)" >> Bukola.fasta
  172  echo "Number of G: $(grep -v '>' GyrBgene2.fasta | grep -o 'G' | wc -l)" >> Bukola.fasta
  173  echo "Number of T: $(grep -v '>' GyrBgene2.fasta | grep -o 'T' | wc -l)" >> Bukola.fasta
  174  echo "Number of C: $(grep -v '>' GyrBgene2.fasta | grep -o 'C' | wc -l)" >> Bukola.fasta
  175  cat Bukola.fasta
  176  history
  177  GC_Content=$(awk "BEGIN {print ($GC_Count / $Total_Count) * 100}")
  178  history
  179  GC_Content=$(awk "BEGIN {print ($GC_Count / $Total_Count) * 100}")
  180  GC_Content=$(awk "BEGIN {print ($GC_Count/$Total_Count) * 100}")
  181  GC_Content=$(awk "BEGIN {print ('$GC_Count / $Total_Count') * 100}")
  182  GC_Content=$(awk "BEGIN {print ($GC_Count / $Total_Count) * 100}")
  183  echo $Total_Count
  184  echo $GC_Count
  185  GC_Count=$(grep -v '>' GyrBgene2.fasta |grep -o [GC] |wc -l)
  186  echo GC_Count
  187  echo $GC_Count
  188  Total_Count=$(grep -v '>' GyrBgene2.fasta |grep -o [AGTC] |wc -l)
  189  echo $Total_Count
  190  GC_Content=$(awk "BEGIN {print ($GC_Count / $Total_Count) * 100}")
  191  echo GC_Content
  192  echo $GC_Content
  193  GC_Contents=$("Scale=2;$GC_Count / $Total_Count) * 100" | bc)
  194  GC_Contents=$("Scale=2; $GC_Count / $Total_Count) * 100")
  195  ls
  196  cat GCcount
  197  rm GCcount wget-log wget-log.1
  198  ls
  199  rm NC_000017.11 'NC_000017.11?report=fasta.1 'NC_000017.11?report=fasta&from=43044295&to=43170327&strand=true' 'NC_000017.11?report=fasta&from=43044295&to=43170327&strand=true ' 'NC_000017.11?report=fasta&from=43044295&to=43170327&strand=true.1'
  200  )
  201  ls
  202  rm  NC_000017.11 
  203  rm 'NC_000017.11?report=fasta.1'
  204  rm 'NC_000017.11?report=fasta&from=43044295&to=43170327&strand=true'
  205  rm 'NC_000017.11?report=fasta&from=43044295&to=43170327&strand=true ' 
  206  rm  'NC_000017.11?report=fasta&from=43044295&to=43170327&strand=true.1'
  207  rm wildtype.gbk.2
  208  ls
  209  rm 'viewer.fcgi?db=nuccore&report=fasta&id=PP072261.1' 
  210  rm 'viewer.fcgi?db=nuccore&report=fasta&id=PP072261.1%0A'
  211  ls
  212  rm GyrBgene.fasta
  213  rm wildtype.gbk.1
  214  ls
  215  history > Bukola.sh
  216  clear
  217  ls
  218  mv *gbk ../Biocomputing
  219  sudo mv wildtype.gbk ../Biocomputing
  220  ls
  221  wget https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.gbk
  222  ls
  223  mv wildtype.gbk ../Biocomputing 
  224  ls -l wildtype.gbk ../Biocomputing 
  225  ls ../Biocomputing 
  226  cd ./Biocomputing/
  227  wget ls
  228  ls
  229  cd ./Bukola/
  230  home
  231  cd 
  232  ls
  233  clear
  234  ls
  235  clear
  236  ls
  237  history
  238  mkdir Charles-Darwin
  239  mkdir biocomputing && cd biocomputing
  240  wget https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.fna https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.gbk https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.gbk
  241  ls
  242  mv *.fna ~/Charles-Darwin/
  243  ls
  244  rm wildtype.gbk.1
  245  ls
  246  grep ‘tatatata’ wildtype.fna 
  247  cd ./Bukola/
  248  cd ../Bukola
  249  grep ‘tatatata’ wildtype.fna 
  250  grep -o ‘tatatata’ wildtype.fna 
  251  cat wildtype.fna 
  252  ls
  253  grep ‘tatatata’ wildtype.fna
  254  history
  255  ls
  256  grep -o "tatatata" wildtype.fna
  257  echo “the wildtype.fna file is mutant”
  258  grep -o "tatatata" wildtype.fna > mutant_lines.txt
  259  cat mutant_lines.txt 
  260  echo “Hemoglobin Subunit Alpha 1”
  261  wget -O HBA1.fasta "https://www.ncbi.nlm.nih.gov/sviewer/viewer.cgi?db=nuccore&report=fasta&id=NC_000016.10"
  262  efetch -db nucleotide -id NC_000016.10 -format fasta > HBA1.fa
  263  grep -v '>' HBA1.fasta | wc -l
  264  cat HBA1.fasta
  265  grep -v '>' HBA1.fasta | wc -l
  266  grep -v “>” HBA1.fasta | wc -l
  267  grep -v ‘>’ HBA1.fasta | wc -l
  268  ls
  269  cat HBA1.fasta
  270  grep -o '>' HBA1.fasta | wc-l
  271  grep -o '>' HBA1.fasta | wc -l
  272  grep -v '>' HBA1.fasta | wc -l
  273  grep -v '>' HBA1.fasta |grep -o 'A' |wc -l
  274  echo “A occurs 22558319 times”
  275  grep -v '>' HBA1.fasta |grep -o 'G' |wc -l
  276  echo “A occurs 18299976 times”
  277  grep -v '>' HBA1.fasta |grep -o 'T' |wc -l
  278  echo "T occurs 22774906"
  279  grep -v '>' HBA1.fasta |grep -o 'C' |wc -l
  280  GC_Counts=$(grep -v '>' HBA1.fasta |grep -o [GC] |wc -l
)
  281  echo $GC_Counts
  282  Total_Counts=$(grep -v '>' HBA1.fasta |grep -o [AGTC] |wc -l)
  283  echo $Total_Counts
  284  echo $GC_Contents
  285  touch Charles-Darwin.fasta
  286  echo "Number of A: $(grep -v '>' HBA1.fasta | grep -o 'A' | wc -l)" >> Charles-Darwin.fasta
  287  echo "Number of G: $(grep -v '>' HBA1.fasta | grep -o 'G' | wc -l)" >> Charles-Darwin.fasta
  288  echo "Number of T: $(grep -v '>' HBA1.fasta | grep -o 'T' | wc -l)" >> Charles-Darwin.fasta
  289  echo "Number of C: $(grep -v '>' HBA1.fasta | grep -o 'C' | wc -l)" >> Charles-Darwin.fasta
  290  echo Charles-Darwin.fasta 
  291  cat Charles-Darwin.fasta 
  292  history > Charles-Darwin.sh
