    1  df
    2  df -H
    3  python
    4  sudo apt-get fastqc
    5  sudo apt-get install fastqc
    6  clear
    7  whoami
    8  pwd
    9  ls
   10  cat README-cloudshell.txt 
   11  mkdir HackBio
   12  ls
   13  ls -lh
   14  clear
   15  cd hack
   16  ckear
   17  clear
   18  cd HackBio/
   19  cat 
   20  ls ../
   21  cd ../
   22  mv HackBio/ hackbio
   23  ls
   24  cd hackbio/
   25  touch hackbio.txt
   26  ls -lh
   27  nano hackbio.txt 
   28  cat hackbio.txt 
   29  ls -lh
   30  less hackbio.txt 
   31  wget https://raw.githubusercontent.com/josoga2/sc_data/main/forensics/forensics.fasta
   32  ls
   33  cp forensics.fasta /..
   34  cp forensics.fasta ../
   35  ls
   36  ls ../
   37  cd ./
   38  cd ../
   39  ls
   40  rm forensics.fasta 
   41  ls
   42  ls hackbio/
   43  cd hackbio/
   44  clear
   45  history 
   46  clear
   47  grep *.fasta
   48  grep sample forensics.fasta 
   49  grep 'sample' forensics.fasta 
   50  grep Sample forensics.fasta 
   51  ls -lh | grep Dec
   52  ls -lh | grep 'Dec'
   53  ls -lh
   54  ls -lh | grep Mar
   55  mkdir bio && cd bio
   56  ls cd ../
   57  cd ../
   58  wc -l forensics.fasta 
   59  mkdir -p ~/miniconda3
   60  wget <https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh> -O ~/miniconda3/miniconda.sh
   61  ls
   62  wget <https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh> -O ~/miniconda3/miniconda.sh
   63  exit()
   64  ls
   65  cd hackbio/
   66  ls
   67  mkdir tiamiyu
   68  mkdir biocomputing && cd biocomputing
   69  wget https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.fna https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.gbk https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.gbk
   70  ls -lh
   71  mv *.fna ../tiamiyu/
   72  ls
   73  rm wildtype.gbk.1 
   74  ls
   75  cdd ../
   76  cd ../
   77  ls
   78  mv tiamiyu/ charles-darwin
   79  ls
   80  cd bio
   81  cd biocls
   82  ls
   83  rm bio
   84  cd ../
   85  ls  -lhbio
   86  ls  -lh bio
   87  rm bio
   88  rm -r bio
   89  ls
   90  ls charles-darwin/
   91  grep 
   92  cd charles-darwin/
   93  ls
   94  cat wildtype.fna 
   95  clear
   96  grep tatata wildtype.fna 
   97  grep tatata wildtype.fna | wc 
   98  grep tatata wildtype.fna 
   99  grep “tatata” wildtype.fna
  100  grep 'tatata' wildtype.fna
  101  grep tatata wildtype.fna | wc 
  102  cat grep ‘tatata’ wildtype.fna
  103  grep ‘tatatata’ wildtype.fna
  104  ls
  105  grep tatata wildtype.fna 
  106  grep tatatata wildtype.fna 
  107  grep 'tatatata' wildtype.fna 
  108  grep tatatata wildtype.fna > mutant-line.txt
  109  ls
  110  cat mutant-line.txt 
  111  k
  112  l
  113  wget https://www.ncbi.nlm.nih.gov/nuccore/NC_000019.10?report=fasta&from=49659572&to=49665857&strand=true
  114  ls
  115  ls -lh
  116  cat wget-log 
  117  rm wget-log 
  118  ls
  119  cat NC_000019.10\?report\=fasta 
  120  clear
  121  ls
  122  rm NC_000019.10\?report\=fasta 
  123  ls
  124  cd hackbio/
  125  ls
  126  charles-darwin/
  127  ls
  128  cd hackbio/charles-darwin/
  129  ls
  130  ls -lh
  131  wget -q -O - "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nuccore&id=NC_000019.10&rettype=fasta"
  132  ls
  133  ls -lh
  134  ls ../
  135  history 
  136  ls
  137  ls ~
  138  ls
  139  ls ~
  140  cd ~
  141  ls
  142  cd bio
  143  cd hackbio/
  144  ls
  145  cd biocomputing/ 
  146  ls
  147  grep tatatata wildtype.gbk 
  148  cd ../charles-darwin/
  149  ls
  150  wget -O gene.fasta "https://www.ncbi.nlm.nih.gov/nuccore/12345?report=fasta"
  151  ls
  152  less gene.fasta 
  153  wget -O irf3.fasta "https://www.ncbi.nlm.nih.gov/nuccore/3661?report=fasta"
  154  ls
  155  wget -O hemoglobin_beta.fasta "https://www.ncbi.nlm.nih.gov/nuccore/3043?report=fasta"
  156  ls -lh
  157  vm hemoglobin_beta.fasta hackbio/charles-darwin/
  158  mv hemoglobin_beta.fasta hackbio/charles-darwin/
  159  ls
  160  cd hackbio/
  161  ls
  162  ls biocomputing/
  163  ls chae
  164  ls charles-darwin/
  165  cd charles-darwin/
  166  rm gene.fasta 
  167  ls -lh
  168  rm irf3.fasta 
  169  ls -lh
  170  wget "https://www.ncbi.nlm.nih.gov/nuccore/3043?report=fasta"
  171  ls -lh
  172  less hemoglobin_beta.fasta 
  173  wget -O hemoglobin_beta.fasta "https://www.ncbi.nlm.nih.gov/sviewer/viewer.cgi?db=nuccore&report=fasta&id=NM_000518.5"
  174  ls -lh
  175  less hemoglobin_beta.fasta 
  176  rm '3043?report=fasta' hemoglobin_beta.fasta 
  177  ls -lh
  178  wget "https://www.ncbi.nlm.nih.gov/sviewer/viewer.cgi?db=nuccore&report=fasta&id=NM_000518.5"
  179  ls -lh
  180  cat 'viewer.cgi?db=nuccore&report=fasta&id=NM_000518.5' 
  181  wget -O hemoglobin_beta.fasta "https://www.ncbi.nlm.nih.gov/sviewer/viewer.cgi?db=nuccore&report=fasta&id=NM_000518.5"
  182  ls
  183  rm 'viewer.cgi?db=nuccore&report=fasta&id=NM_000518.5' 
  184  ls
  185  wc -l hemoglobin_beta.fasta 
  186  wc hemoglobin_beta.fasta 
  187  help -wc
  188  wc help
  189  wc?
  190  wc hemoglobin_beta.fasta 
  191  wc -l
  192  wc -l hemoglobin_beta.fasta 
  193  wc -l hemoglobin_beta.fasta |-n 10
  194  wc -l hemoglobin_beta.fasta
  195  wc -l hemoglobin_beta.fasta - 1
  196  wc hemoglobin_beta.fasta 
  197  wc -l hemoglobin_beta.fasta | sort -10
  198  sort --help 
  199  wc -l hemoglobin_beta.fasta |-n 1
  200  cat hemoglobin_beta.fasta 
  201  grep --help 
  202  grep -v '^>'
  203  grep -v '^>' hemoglobin_beta.fasta 
  204  cat hemoglobin_beta.fasta 
  205  tail -n 1 hemoglobin_beta.fasta 
  206  grep -v '^>' hemoglobin_beta.fasta 
  207  grep -v '^>' hemoglobin_beta.fasta | head -n -1 | wc -l
  208  ls
  209  cd hackbio/biocomputing
  210  ls
  211  cd ../
  212  ls
  213  cd charles-darwin/
  214  ls
  215  wc grep A hemoglobin_beta.fasta 
  216  grep A hemoglobin_beta.fasta | wc
  217  grep 'A' hemoglobin_beta.fasta | wc
  218  wc hemoglobin_beta.fasta 
  219  grep A hemoglobin_beta.fasta 
  220  grep A hemoglobin_beta.fasta | wc
  221  grep -o A hemoglobin_beta.fasta 
  222  grep -o A hemoglobin_beta.fasta | wc -l
  223  grep -o A hemoglobin_beta.fasta | wc -l #140 times
  224  grep -o C hemoglobin_beta.fasta | wc -l #140 times
  225  grep -o G hemoglobin_beta.fasta | wc -l #140 times
  226  grep -o T hemoglobin_beta.fasta | wc -l #140 times
  227  ls
  228  cd hackbio/
  229  ls
  230  echo Homo sapiens hemoglobin subunit beta (HBB)
  231  echo Homo sapiens hemoglobin subunit beta (HBB)'
  232  echo 'Homo sapiens hemoglobin subunit beta (HBB)'
  233  # 'Homo sapiens hemoglobin subunit beta (HBB)'
  234  #'Homo sapiens hemoglobin subunit beta (HBB)'
  235  touch charles-darwin.fa
  236  echo >> charles-darwin.fa 
  237  cat charles-darwin.fa 
  238  nano charles-darwin.fa 
  239  mv charles-darwin
  240  ls
  241  mv charles-darwin.fa charles-darwin
  242  ls
  243  cd charles-darwin/
  244  ls
  245  echo 'the following into the file using' >> charles-darwin.fa 
  246  cat charles-darwin.fa 
  247  nano charles-darwin.fa 
  248  cat charles-darwin.fa 
  249  ls
  250  cd hackbio/charles-darwin/
  251  ls
  252  cat mutant-line.txt 
  253  grep -o G hemoglobin_beta.fasta | wc -l
  254  echo grep -o G hemoglobin_beta.fasta | wc -l >> charles-darwin.fa 
  255  cat charles-darwin.fa 
  256  nano charles-darwin.fa 
  257  echo grep -o G hemoglobin_beta.fasta | wc -l >> charles-darwin.fa 
  258  cat charles-darwin.fa 
  259  grep -o G hemoglobin_beta.fasta | wc -l
  260  $n_G=grep -o G hemoglobin_beta.fasta | wc -l
  261  echo 'The number of A:140' > charles-darwin.fa 
  262  cat charles-darwin.fa 
  263  echo 'The number of G:165' >> charles-darwin.fa 
  264  echo 'The number of T:167' >> charles-darwin.fa 
  265  echo 'The number of C:157' >> charles-darwin.fa 
  266  cat charles-darwin.fa 
  267  history
  268  history >> charles-darwin.sh
  269  ls
  270  cat charles-darwin.sh 
  271  clear
  272  history 
  273  ls
  274  mv charles-darwin.fa output/
  275  mkdir output && mv charles-darwin.fa output
  276  mkdir script && mv charles-darwin.sh script
  277  ls
  278  cat script/charles-darwin.sh 
  279  cat output/charles-darwin.fa 
  280  ls ~/hackbi
  281  cd ~
  282  ls
  283  cd hackbio/
  284  ls
  285  ls biocomputing/ charles-darwin/
  286  ls
  287  cd hackbio/charles-darwin/
  288  ls
  289  grep -o -i '[GC]' hemoglobin_beta.fasta | wc -l
  290  grep G hemoglobin_beta.fasta | wc -l
  291  grep -o  G hemoglobin_beta.fasta | wc -l
  292  grep -o  C hemoglobin_beta.fasta | wc -l
  293  grep -o -i '[GC]' hemoglobin_beta.fasta 
  294  cat hemoglobin_beta.fasta 
  295  grep --help 
  296  grep -o '[GC]' hemoglobin_beta.fasta 
  297  grep -o '[GC]' hemoglobin_beta.fasta | wc -l
  298  clear
  299  GC_count=$(grep -o -i '[GC]' hemoglobin_beta.fasta | wc -l)
  300  GC_count=$(grep -o '[GC]' hemoglobin_beta.fasta | wc -l)
  301  total_bases=$(grep -o '[ATGC]' hemoglobin_beta.fasta | wc -l)
  302  bc --help
  303  GC_content=$(echo "scale=2; $GC_count / $total_bases * 100" | bc)
  304  wc --help 
  305  GC_content=$(awk "BEGIN {print ($GC_count / $total_bases) * 100}")
  306  echo "GC content: $GC_content%"
  307  history 
  308  history script/charles-darwin.sh 
  309  history >  script/charles-darwin.sh 
